### PR TITLE
Problem: no ADR document for feemarket module

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Comment by Muggle-Du** [comment]
LGTM

**Comment by adu-crypto** [comment]
LGTM

**Review by yihuang — APPROVED** [review]

File: docs/architecture/adr-003.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+<!-- > This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future. -->
+
+Requires a major network upgrade. 
+Local validator won't be able to choose arbitrary which fee value they would like to accept after the upgrade, but they will be able to define a local "tip" that can be added on top of this basefee. 
+
+### Backwards Compatibility
+
+<!-- All ADRs that introduce backward incompatibilities must include a section describing these incompatibilities and their severity. The ADR must explain how the author proposes to deal with these incompatibilities. ADR submissions without a sufficient backward compatibility treatise may be rejected outright. -->
+
+The proposed ADR is not backward compatible.
```
Comment (tomtau) [inline docs/architecture/adr-003.md:46]: I assume "legacy transactions" will still be accepted, or will they not? https://ethereum.stackexchange.com/questions/111654/how-legacy-transaction-are-handled-after-eip1559-update

**Review by tomtau — APPROVED** [review]
looks good, it may be worth adding a few comments regarding Web3. I assume it'll be the same as with Ethereum, i.e. new Web3 client will be able to make use of it (e.g. they can query the base fee and construct the new transaction types), while the old ones will still work as before (but won't make use of the new pricing mechanism, i.e. will overpay).
Is that the case, or is there something Ethermint-specific?

File: docs/architecture/adr-003.md (hunk 1 above)
Comment (thomas-nguy) [reply docs/architecture/adr-003.md]: Yes you are right, I was thinking about cosmos transaction but at web3 level, it is backward compatible with legacy txs. 

I have fixed this section to add more clarity

**Pull request merged**